### ISSUE TITLE
Extension documentation needs some updating

### ISSUE THREAD

**Issue by hmelman** [minad/corfu#167]
I'm finally configuring corfu (and liking it).  I think the documentation for the extensions needs to be improved a bit. The commentary section in each file needs to be updated and should consistently describe how to enable them. `corfu-history` is the best but I think it should explicitly say:

``` elisp
;; Remember and sort recently used candidates at the top
(corfu-history-mode)
;; Optionally use savehist-mode to remember these across emacs sessions
(savehist-mode)
(add-to-list 'savehist-additional-variables 'corfu-history)
```

The `corfu-indexed` commentary should mention enabling `(corfu-indexed-mode)`.  It talks about accessing candidates via a prefix argument but it doesn't say as a prefix to which command.  It wasn't hard to guess `RET` (which I gather is `corfu-insert`) but it would be nice to be explicit.  Also I wonder if there's a way to just let some number key work without a `RET`.  Maybe bind `M-NUMBER` to immediate selection?  Or use this idea to introduce `corfu-quick` (though I believe that still needs a command used before entering a number).

The `corfu-quick` commentary mentions `corfu-quick-exit` which isn't defined and doesn't mention all the autoloaded commands.

The `corfu-info` commentary should name the two commands defined and perhaps suggest bindings, or at least suggest what keymap to bind them in.  I'm not a company user and I don't really understand what information these commands display (is it more than for just elisp docs, is it whatever eldoc displays? and location is any definition that xref could find?) or when.  I guess I have to select a candidate and then invoke one of these commands.  

I think in the README should list these extensions with `corfu-indexed` and `corfu-quick` next to each other as they're kinda of alternatives.  Also it should be more explicit about configuring them or should say "See the Commentary sections of these files to configure them".

**Comment by minad** [comment]
> Also I wonder if there's a way to just let some number key work without a RET. Maybe bind M-NUMBER to immediate selection?

I agree agree that we could improve the situation with regarding to direct candidate selection with hotkeys, but this would likely require a separate extension. I've not yet figured out how to consolidate that with corfu-quick and corfu-indexed, since it would be a bit odd to have three different quick selection extensions. See https://github.com/minad/vertico/issues/218 for the related issue on the Vertico tracker. In any case we should implement the same in Vertico.

**Comment by hmelman** [comment]
On May 13, 2022, at 2:41 PM, Daniel Mendler ***@***.***> wrote:
> 
> 
> Closed #167 via 443905d.

The changes look good.  

One question: Since the functions in corfu-info are autoloaded and bound in corfu-map (so there's no user needed config to enable them) they're not really extensions, they're just features that happen to be in another file.  Are they separate because they're experimental?

Howard

**Comment by minad** [comment]
> One question: Since the functions in corfu-info are autoloaded and bound in corfu-map (so there's no user needed config to enable them) they're not really extensions, they're just features that happen to be in another file.  Are they separate because they're experimental?

You are right about that. The info commands are bound by default for backward compatibility since the corfu-info extension was moved out of corfu.el. But corfu.el would function perfectly well without it, you just don't have access to this auxiliary functionality then. I alsYou are right about that. The info commands are bound by default for backward compatibility since the corfu-info extension was moved out of corfu.el. But corfu.el would function perfectly well without it, you just don't have access to this auxiliary functionality then. I use these commands only rarely and some people prefer corfu-doc instead.
